src/App.js

```diff
@@ -1,15 +1,32 @@
 import React from 'react';
+import { connect } from 'react-redux'
+import { Switch, Route, Redirect } from 'react-router-dom';
 import './App.css';
-import BookList from './components/booklist/booklist.component';
-import BookForm from './components/book-form/book-form.component';
-
-function App() {
-  return (
-    <div>
-      <BookList />
-      <BookForm />
-    </div>
-  );
+import Homepage from './pages/homepage/homepage';
+import SignInUp from './pages/sign-in-up/sign-in-up';
+
+class App extends React.Component {
+
+  	render() {
+	  	return (
+	  	  <Switch>
+	      	<Route exact path='/' render = {() => 
```

Comment:
Additionally there's a way to do this sort of logic. We normally use "higher order components" to conditionally render a specific component depending on state. I wouldn't do my conditonal here in the Route